libssh.map

```diff
@@ -0,0 +1,425 @@
+# When new symbols are added in a release, they must be added as:
```

Comment:
nit: it might be better to place this file under `src/` as it is only referred to from `src/CMakeList.txt`?

Reply:
Yes, it makes sense to store it in ``src/``. My only concern is about future maintenance of the linker map that have to be done in each release. Storing it in ``src/`` can make the maintainer to forget to do it. I think I should add rules in the ``Makefile`` to check the ABI (using ``abi-compliance-checker``)